src/main/java/com/google/devtools/build/lib/analysis/LocationExpander.java

```diff
@@ -362,9 +362,22 @@ private String functionName() {
       }
     }
 
+    boolean inAspect = ruleContext.getAspects().size() > 0;
+
     // Add all destination locations.
     for (OutputFile out : ruleContext.getRule().getOutputFiles()) {
-      mapGet(locationMap, out.getLabel()).add(ruleContext.createOutputArtifact(out));
+      Artifact artifact;
+      if (inAspect) {
+        // In aspect processing, so don't create & verify a new artifact, just find any cached ones.
+        artifact = ruleContext.getOutputArtifactsProvider().getOutputArtifact(out);
+        // FIXME: I see correct output if I just ignore missing artifacts here. Is that problematic?
```

Comment:
Hm. In what case does that happen? Naively, I'd think that `getOutputArtifactsProvider()` would return an `Artifact` for each `OutputFile` in `getOutputFiles()`.